src/v1/combo-chart/combo-chart.ts

```diff
@@ -0,0 +1,669 @@
+  const yAxisColumns = configDimensions?.[1].columns ?? [];
+  const legend = configDimensions?.[2]?.columns[0] ?? [];
+  const type = inputType.split("-")[0];
+  const isStacked = inputType.split("-").length > 1;
+
+  return {
+    getLabels: () => getDataForColumn(xAxisColumns[0], dataArr),
+    getDatasets: () => {
+      if (!_.isEmpty(legend)) {
+        return _.map(getSeriesDataForColumn(legend, dataArr), (col, idx) => {
+          return {
+            label: `${col}- ${inputType}`,
+            data: getDataForColumnForSeries(yAxisColumns[0], dataArr, col),
+            type: `${type}`,
+            // yAxisID: `${type}-y${idx.toString()}`,
+            stack: `${type}-x0${isStacked ? "-stacked" : "y" + idx.toString()}`,
+            yPos: idx,
+            backgroundColor: '',
+            borderColor: '',
+            datalabels: {
+              anchor: "end",
+            },
+          };
+        });
+      } else {
+        return _.map(yAxisColumns, (col, idx) => {
+          return {
+            label: `${col.name}- ${inputType}`,
+            data: getDataForColumn(col, dataArr),
+            //yAxisID: `${type}-y${idx.toString()}`,
+            stack: `${type}-x0${isStacked ? "-stacked" : "y" + idx.toString()}`,
+            type: `${type}`,
+            backgroundColor: '',
+            borderColor: '',
+            yPos: idx,
+            datalabels: {
+              anchor: "end",
+            },
+          };
+        });
+      }
+    },
+
+    getPointDetails: (xPos: number, yPos: number): PointVal[] => {
+      console.log(type);
+      if (!_.isEmpty(legend)) {
+        return [
+          {
+            columnId: legend.id,
+            value: getDataForColumn(legend, dataArr)[yPos],
+          },
+          {
+            columnId: xAxisColumns[0].id,
+            value: getDataForColumn(xAxisColumns[0], dataArr)[xPos],
+          },
+          {
+            columnId: yAxisColumns[0].id,
+            value: getDataForColumn(yAxisColumns[0], dataArr)[xPos],
+          },
+        ];
+      } else {
+        return [
+          {
+            columnId: xAxisColumns[0].id,
+            value: getDataForColumn(xAxisColumns[0], dataArr)[xPos],
+          },
+          {
+            columnId: yAxisColumns[yPos].id,
+            value: getDataForColumn(yAxisColumns[yPos], dataArr)[xPos],
+          },
+        ];
+      }
+    },
+  };
+}
+
+function getDataModel(chartModel: ChartModel) {
+  const xColumnDimension =
+    chartModel.config?.chartConfig?.[0].dimensions.filter(
+      (dim) => dim.key === "x"
+    )[0];
+  // column chart model
+  const columnChartModel = getChartDataModel(
+    chartModel.config?.chartConfig?.[0].dimensions ?? [],
+    chartModel.data?.[0].data ?? ([] as any),
+    COMBO_CHART_TYPE.bar
+  );
+
+  // line chart model
+  const lineChartModel = getChartDataModel(
+    [xColumnDimension, ...chartModel.config?.chartConfig?.[1].dimensions] ?? [],
```

Comment:
lets fetch config by name and not by index. that can be misleading.

Reply:
updated